homeassistant/components/motioneye/__init__.py

```diff
@@ -429,6 +468,149 @@ async def handle_webhook(
     return None
 
 
+class MotionEyeServices:
+    """Class that holds motionEye services that should be published to hass."""
+
+    SCHEMA_TEXT_OVERLAY = vol.In(
+        [
+            KEY_TEXT_OVERLAY_DISABLED,
+            KEY_TEXT_OVERLAY_TIMESTAMP,
+            KEY_TEXT_OVERLAY_CUSTOM_TEXT,
+            KEY_TEXT_OVERLAY_CAMERA_NAME,
+        ]
+    )
+
+    SCHEMA_DEVICE_OR_ENTITIES = {
+        vol.Optional(ATTR_DEVICE_ID): vol.All(cv.ensure_list, [cv.string]),
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    }
+
+    SERVICE_TO_ACTION = {
+        SERVICE_SNAPSHOT: KEY_ACTION_SNAPSHOT,
+    }
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize with hass object."""
+        self._hass = hass
+
+    @callback
+    def async_register(self) -> None:
+        """Register all our services."""
+        self._hass.services.async_register(
```

Comment:
Why not use entity services?

<https://developers.home-assistant.io/docs/dev_101_services/#entity-services>